src/BenchmarkDotNet.Diagnostics.Windows/Tracing/NativeMemoryLogParser.cs

```diff
@@ -23,18 +27,32 @@ public class NativeMemoryLogParser
 
         private readonly ILogger logger;
 
-        public NativeMemoryLogParser(string etlFilePath, BenchmarkCase benchmarkCase, ILogger logger)
+        private readonly (string module, Regex function)[] benchmarkedFunctionStacks;
+
+        public NativeMemoryLogParser(string etlFilePath, BenchmarkCase benchmarkCase, ILogger logger,
+            string programName)
         {
             this.etlFilePath = etlFilePath;
             this.benchmarkCase = benchmarkCase;
             this.logger = logger;
+
+            var module = Path.GetFileNameWithoutExtension(benchmarkCase.Descriptor.WorkloadMethod.Module.Name);
+            var function = new Regex($@"{benchmarkCase.Descriptor.WorkloadMethod.DeclaringType?.FullName}\.{benchmarkCase.Descriptor.WorkloadMethod.Name}\(");
+
+            benchmarkedFunctionStacks = new (string module, Regex function)[]
+            {
+                (module, function),
+                (programName, new Regex(@"BenchmarkDotNet\.Autogenerated\.Runnable_[0-9]+.WorkloadActionNoUnroll\(")),
```

Comment:
nit: use `RegexOptions.Compiled` to get the best perf

```suggestion
                (programName, new Regex(@"BenchmarkDotNet\.Autogenerated\.Runnable_[0-9]+.WorkloadActionNoUnroll\(", RegexOptions.Compiled)),
```